packages/graphile-build-pg/src/plugins/PgIntrospectionPlugin.js

```diff
@@ -448,6 +537,8 @@ export default (async function PgIntrospectionPlugin(
       pgClient.on("error", e => {
         debug("pgClient error occurred: %s", e);
       });
+
+      //TODO why the if(pgClient) checks here and not above
```

Comment:
I guess because above they are guaranteed to exist; here runs after a period of time, which might be after this is called: https://github.com/jclangst/graphile-build/blob/9bc063ba1323b8ac1a40276ffac37b5c0098fd8d/packages/graphile-build-pg/src/plugins/PgIntrospectionPlugin.js#L521-L524

Reply:
I'm thinking that because it looks like a lot of thought went into getting the ``pgClient`` based on the polymorphism of ``pgConfig`` in the ``withPgClient`` utility, we might want to stick to using that. That way if there ever needs to be changes in how pgConfig is handled, it will remain consistent across all plugins.

Reply:
That would have to be done *very* carefully to ensure that no functionality changed here. PostGraphile is not the only user of graphile-build-pg. `withPgClient` was generally intended for short-lived clients (a few statements and then release) whereas this is a long-lived client that listens for NOTIFY events.